src/main/rx/crsf.c

```diff
@@ -45,8 +45,8 @@
 
 #include "telemetry/crsf.h"
 
-#define CRSF_TIME_NEEDED_PER_FRAME_US   1000
-#define CRSF_TIME_BETWEEN_FRAMES_US     4000 // a frame is sent by the transmitter every 4 milliseconds
+#define CRSF_TIME_NEEDED_PER_FRAME_US   1100 // 700 ms + 400 ms for potential ad-hoc request
+#define CRSF_TIME_BETWEEN_FRAMES_US     6667 // At fastest, frames are sent by the transmitter every 6.667 milliseconds, 150 Hz
```

Comment:
Crossfire specification says minimum time between frames is 4ms. Why has this been changed?

Reply:
@martinbudden Based on the spec, the 4ms update rate requirement is for the RC to the Crossfire TX, not from the RX to the flight controller.  The 6667 update rate is based on the fastest case scenario of 150Hz, which will also scale fine for 50Hz due to the static frame buffer.

Reply:
This udpate rate was determined by scope measurements and was not just arbitrarily specified.

Reply:
This is were there is a difference between current behaviour and the spec. It may be true that current frames are sent at 150Hz, but who is to say it won't increase in future?

Reply:
We can revert it if you'd like, as I can't see where this refresh rate really has much effect.

The spec doesn't state what the requirement is for the RX frame rate, other than the baud rate, parity, etc, so I'm not sure how this behavior can be compared to the spec, persay:

"Dual wire / full duplex UART

This configuration is usually used on the flying platform side. Two devices are connected by
regular UART connection. Only non-inverted ( regular ) UART is supported in this
configuration. The UART runs at 400kbaud 8N1 at 3.0 to 3.3V level. "

It only specifies the 4ms timing with respect to the communication between the transmitter and the crossfire TX:

"Single wire half duplex UART


This configuration is usually used between RC and Crossfire TX. XF TX support both
inverted and non-inverted UART. The RC acts as master in this case and the XF TX can
only send telemetry if it’s synchronized to the RC frames sent by the RC. The RC frame
update rate should not be sent more often than every 4ms. The UART runs at 400kbaud
8N1 ( inverted or non-inverted ) at 3.3V level."

This is required because the Crossfire TX uses a single pin for half duplex communication with the transmitter.

The only place I saw it really being required on the RX code is for resetting the frame position and for half duplex.  Since half duplex is not available on the RX side, it won't be needed for that purpose.  The frame position is also maintained when the end of a frame is sensed by the frame length (https://github.com/betaflight/betaflight/blob/master/src/main/rx/crsf.c#L149) ... this is required, as the RX "piggy-backs" other non-RC related incoming messaging (MSP) immediately after the RC frame.  

As long the ISR function is called properly and the frame is received at full length, the frame position should be reset immediately in preparation for the next frame.  If the frame rate were to increase, this position reset should accommodate that change naturally. However, if you'd like to shorten this time to offset it from the expected rate, we can revert it.  I have no problem with it :)

Reply:
If I'm understanding your point correctly, you're basically stating that since the TX module can obtain data at that rate, the RX should accomodate it in the event the frames are processed at the same rate as they are sent to the TX from the transmitter?